zenoh/src/net/routing/dispatcher/resource.rs

```diff
@@ -306,6 +307,7 @@ impl Resource {
                         }
                     }
                 }
+                mutres.nonwild_prefix.take();
```

Comment:
This should not be necessary as mutes will be dropped.

Reply:
mutres is a reference here, so no actual drop operation will happen.  Based on my tests, this line is necessary, since non_wild prefix might contain an Arc to a parent, that will fail to clean due to higher than expect Arc reference count. I.e. we can have following situation: mutres.non_wild prefix holds parent, grandparent.childs hold parent, mutres.parent holds parent, together with reclone it will lead to Arc::strong_count of 4, and thus we will not be able to clean the parent.

Reply:
Well that was the point of the code. You can't clean a resource that has childs. You have to clean the childs first.

Reply:
But we call clean(parent) just after and so need to clean nonwild_prefix Arc before that. Ok